### DISCUSSION TITLE
How to configure 'No loop' in v5?

### DISCUSSION THREAD

**Discussion by GeronimoCH** [fancyapps/ui#572] [Q&A] [answered]
I have a gallery of images. If I click 'Next' the next image is displayed. 'Next' at the last image of the gallery displays the the first one. In fancybox v3 there was a configuration attribute 'loop' that could be set to 'false' to stop 'next' at the last image (or 'prev' at the fist image). In fancyapps v5 I can not find an equivalent configuration option. How can I configure 'no loop'?

Thanks for any hints.
Hannes

**Comment by fancyapps** [comment, marked as answer]
Hi,

Fancybox is built around Carousel component, therefore you simply use `infinite` option, like this:

```js
Fancybox.bind('[data-fancybox="gallery"]', {
  Carousel : {
    infinite: false
  }
}); 
```

https://jsfiddle.net/srz68gx3/

**Reply by GeronimoCH** [reply]
Hi Janis

Thanks a lot for the quick reply. I admit, I did not look at the carousel options.

Hannes

**Comment by fanyspania** [comment]
I have a gallery with fancybox v5. Please help me! I need when i press button class="f-button" title="Toggle slideshow" data-fancybox-toggle-slideshow, start a audio file from html.

**Reply by fancyapps** [reply]
@fanyspania Hi, 1) How to know when slideshow starts - use `startSlideshow` event, see https://fancyapps.com/fancybox/api/events/ 2) How to access DOM element of the current slide - use `Fancybox.getSlide().el`. Combine this and you will be able to accomplish your task. Good luck! :)

**Reply by fanyspania** [reply]
Thank you very much.

**Reply by fanyspania** [reply]
I tried, but I think I'm wrong. This is my code:
 on: {
	    startSlideshow: (fancybox, slide) => {
	      // The content of this slide is loaded and ready to be revealed
	      Fancybox.getSlide(1).el = "audio.mp3";
	    },
	  },
Can you write me the correct code, please.

**Reply by fancyapps** [reply]
1) Why do you pass `1` to `Fancybox.getSlide()` ? That method does not accept any argument. 
2) What is the meaning of `Fancybox.getSlide(1).el = "audio.mp3";`? I already explained you that you use that to `to access DOM element of the current slide`. Do you understand what is "DOM element"? Your audio element is probably somewhere inside that element, so just get reference to it (using `querySelector`) and then that do whatever your need.

**Reply by fanyspania** [reply]
I am a beginner in web programming. I really like fancybox and want to add it to my web page. Please help me to be able to implement an audio when I change to automatic slideshow. Thank you very much.

**Reply by fancyapps** [reply]
Sorry, but your question is non-answerable, I do not know nothing about your page, your code, your requirements, etc. If you had created a jsfiddle demo, I would have already forket it and provided you with a solution.

**Reply by fanyspania** [reply]
Thanks bro.

**Reply by fanyspania** [reply]
In the end I was able to solve it. Thank you very much and I am adding the code that starts an audio when the slideshow starts. 
Maybe someone needs it.
<script>
    Fancybox.bind('[data-fancybox]', {
        on: {
            startSlideshow: (fancybox) => {
                var currentSlide = Fancybox.getSlide().el; 

                var audioElement = document.getElementById('miaudio');
                audioElement.play();
            },
            endSlideshow: (fancybox) => {

                var audioElement = document.getElementById('miaudio');
                 audioElement.pause();
            },
        },
        Thumbs : {
          showOnStart: false
        },
        Slideshow: {
          playOnStart: false,
          timeout:4000
        },
    });
</script>
